client/lib/js/react/components/ledgerplugin.js

```diff
@@ -0,0 +1,142 @@
+            console.warn('already connected, ignoring connection request');
+            return Promise.resolve(null);
+        }
+
+        const streamUri = accountUri + '/changes';
+        console.log('subscribing to ' + streamUri);
+
+        const reconnect = reconnectCore(() => {
+            return new SimpleWebsocket(streamUri);
+        });
+
+        return new Promise((resolve, reject) => {
+            this.connection = reconnect({immediate: true}, (ws) => {
+                ws.on('open', () => {
+                    console.log('ws connected to ' + streamUri);
+                })
+                ws.on('data', (msg) => {
+                    const notification = JSON.parse(msg);
+                    co.wrap(this._handleNotification)
+                        .call(this, notification)
+                        .catch((err) => {
+                            console.warn('failure while processing notification: ' +
+                            (err && err.stack) ? err.stack : err)
+                        });
+                });
+                ws.on('close', () => {
+                    console.log('ws disconnected from ' + streamUri);
+                });
+            })
+                .once('connect', () => resolve(null))
+                .on('connect', () => {
+                    this.connected = true;
+                    this.emit('connect');
+                })
+                .on('disconnect', () => {
+                    this.connected = false;
+                    this.emit('disconnect');
+                })
+                .on('error', (err) => {
+                    console.warn('ws error on ' + streamUri + ': ' + err);
+                    reject(err);
+                })
+                .connect();
+        });
+    }
+
+    disconnect() {
+        if (this.connection) {
+            this.connection.disconnect();
+            this.connection = null;
+        }
+    }
+
+    isConnected() {
+        return this.connected;
+    }
+
+    validateTransfer(transfer) {
+        // validator.validate('TransferTemplate', transfer)
+    }
+
+    getInfo() {
+    }
+
+    getAccount() {
+        return this.credentials.account;
+    }
+
+    getBalance() {
+        return co.wrap(this._getBalance).call(this);
+    }
+
+    * _getBalance() {
+        const creds = this.credentials;
+        let res;
+        try {
+            res = yield request('assets_for_account', {
+                urlTemplateSpec: {
+                    accountId: creds.account.id
+                }
+            });
+        } catch (e) {
+            throw new Error('Unable to determine current balance');
+        }
+        res = res.assets.bigchain.length;
+        console.log('balance: ', res);
+        return res;
+    }
+
+
+    getConnectors() {
+    }
+
+    send(transfer) {
+        return co.wrap(this._send).call(this, transfer);
+    }
+
+    * _send(transfer) {
+        
+    }
+
+    fulfillCondition() {
+    }
+
+    replyToTransfer() {
```

Comment:
it's a bootstrap for the next PR, where I will create the actual interledger example